specs/commandline/key.md

```diff
@@ -0,0 +1,132 @@
+# notation key
+
+## Description
+
+Use ```notation key``` command to manage keys used for signing. User can add/update/list/remove key to/from signing key list. Please be noted this command doesn't manage the lifecycle of signing key itself, it manages the signing key list only.
+
+## Outline
+
+### notation key command
+
+```text
+Manage keys used for signing
+
+Usage:
+  notation key [command]
+
+Available Commands:
+  add         Add key to signing key list
+  list        List keys used for signing
+  delete      Delete key from signing key list
+  update      Update key in signing key list
+
+Flags:
+  -h, --help  help for key
+```
+
+### notation key add
+
+```text
+Add key to signing key list
+
+Usage:
+  notation key add [flags] <key_name> [key_path cert_path]
```

Comment:
By exposing this command with `[key_path cert_path]` we are saying, notation support location with unprotected keys only.

If we are saying we dont support signing with local keys we should remove [`key_path cert_path]`. If we are supporting local signing we should add support for accepting password.